src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptions.cs

```diff
@@ -25,13 +35,42 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 throw new Exception("The selected persistence doesn't have support for subscription storage. Select another persistence or disable the message-driven subscriptions feature using endpointConfiguration.DisableFeature<MessageDrivenSubscriptions>()");
             }
 
-            context.Pipeline.Register<SubscriptionReceiverBehavior.Registration>();
-            var authorizer = context.Settings.GetSubscriptionAuthorizer();
-            if (authorizer == null)
+            var transportInfrastructure = context.Settings.Get<TransportInfrastructure>();
+            var canReceive = !context.Settings.GetOrDefault<bool>("Endpoint.SendOnly");
+            var conventions = context.Settings.Get<Conventions>();
+            var enforceBestPractices = context.Settings.Get<bool>(BestPracticesFeature.EnforceBestPracticesSettingsKey);
```

Comment:
Had one more chat with @timbussmann and @danielmarbach and we decided to leave this as is for now.

Seems like "best practices" is more of a generic thing and that the core would provide "conventions" that features could use to enforce things like "not publishing a command".

In short: Instead of pulling best practices to a feature have relevant features use the conventions to perform checks relevant for their specific functionality.  

Eg. since routing is dealing with how to "reply to a message" it would check that the message isn't a command or event.